engine/test/test_path.cpp

```diff
@@ -0,0 +1,111 @@
+/* Copyright (C) 2003-2017 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>.  */
+
+#include "gtest/gtest.h"
+
+#include "prefix.h"
+#include "util.h"
+
+struct TestData
+{
+    const char *path;
+    const char *dir;
+    const char *base;
+};
+
+static const TestData kUnixTestData[] =
+{
+    { "/", "/", "" },
+    { "/foo", "/", "foo" },
+    { "/foo/", "/foo", "" },
+    { "/foo/bar", "/foo", "bar" },
+    { "/foo/bar/", "/foo/bar", "" },
+    { "", "", "" },
+    { "foo", "", "foo" },
+    { "foo/", "foo", "" },
+    { "foo/bar", "foo", "bar" },
+    { "foo/bar/", "foo/bar", "" },
```

Comment:
I've gone for the latter semantic in terms of trailing slashes - essentially, trailing slashes are trimmed from dir unless they are required as part of an absolute specifier. On UNIX paths, this just means trim all trailing slashes except one in position 0. On Win32, it's trim trailing slashes down to the end of an absolute (drive or UNC) base.

In terms of '/0', these path functions are string based - so \0 should just be another character. I think it would be the role of 'path object' to check for validity and such, rather than functions which process paths as strings (which might suggest we should not trim trailing slashes).